packages/autocomplete-core/src/__tests__/getSources.test.ts

```diff
@@ -110,4 +111,25 @@ describe('getSources', () => {
 
     expect(onStateChange.mock.calls.pop()[0].state.collections).toHaveLength(2);
   });
+
+  test('with nothing returned from getItems throws', async () => {
+    const spy = jest.spyOn(handlers, 'onInput');
+
+    const { inputElement } = createPlayground(createAutocomplete, {
+      getSources() {
+        return [createSource({ sourceId: 'source1', getItems: () => {} })];
+      },
+    });
+
+    await expect(() => {
+      inputElement.focus();
+      userEvent.type(inputElement, 'a');
+
+      return spy.mock.results[0].value;
+    }).rejects.toThrow(
+      '[Autocomplete] The `getItems` function must return an array of items but returned type "object":\n\n[\n  null\n]'
```

Comment:
`[null]` is misleading here since users would expect to see `undefined`.

See suggestion below.